### DISCUSSION TITLE
[vaultwarden::api::notifications][ERROR] when trying to enable websocke

### DISCUSSION THREAD

**Discussion by trapassati** [dani-garcia/vaultwarden#1952] [Q&A]
**Subject of the issue**

[vaultwarden::api::notifications][ERROR] when trying to enable websocket
I couldn’t find the same error documented before.

**Deployment environment**

Deployed on docker (vaultwarden/server:latest)

`WEBSOCKET_ENABLED is set to “true”`

**Expected behaviour**

Expect websocket to work. For example, for a security note to sync immediately between web vault and extensions.

**Actual behaviour**

Doesn’t work.

**Troubleshooting data**
I use docker exec command into the contain

```
$ docker exec -i -t bitwarden curl 127.0.0.1:3012

WebSocket Protocol Error: Unable to parse WebSocket key
```

**Comment by BlackDex** [comment]
Please read: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-WebSocket-notifications

**Reply by trapassati** [reply]
Thanks, I've already read it a dozen times, I've configured the reverse proxy, but it still doesn't work

This is my ngnix config


```
  location /notifications/hub {
    proxy_pass http://192.168.10.2:3012;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
  }
  
  location /notifications/hub/negotiate {
    proxy_pass http://192.168.10.2:80;
  }
```

**Comment by BlackDex** [comment]
If with doesn't work you mean that curl responds with an error, that is correct. Since curl doesn't understand websockets (at least not by default as far as i know)

**Reply by trapassati** [reply]
I used curl for testing.

In any case it doesn't even work from mydomian.com/notifications/hub/

I have the same error:

`WebSocket Protocol Error: Unable to parse WebSocket key.`

**Comment by BlackDex** [comment]
Websockets aren't supposed to be accessed directly via the browser.
It is used by the web-vault and desktop and extensions to get sync updates directly.

To test it, open two different browsers, Firefox and chromium for example. Open the web-vault on both, add a new item in one and it should appear in the other almost instantly

**Reply by trapassati** [reply]
I've tried to open the web version in two different browser and the live sync not wok

Anyway, same error in valutwarden log

```
[2021-08-30 14:36:11.011][parity_ws][INFO] Listening for new connections on 0.0.0.0:3012.,
[2021-08-30 14:36:11.025][start][INFO] Rocket has launched from http://0.0.0.0:80,
[2021-08-30 14:36:19.664][vaultwarden::api::notifications][ERROR] ,
    ###########################################################,
    '/notifications/hub' should be proxied to the websocket server or notifications won't work.,
    Go to the Wiki for more info, or disable WebSockets setting WEBSOCKET_ENABLED=false.,
    ###########################################################################################,
,
```

WebSocket isn't working

**Reply by BlackDex** [reply]
Then the proxy is configured incorrectly.
Please check the proxy examples again.
